### DISCUSSION TITLE
MCO Home A8-9 and Z-stick Gen5+

### DISCUSSION THREAD

**Discussion by nkatsaros443** [zwave-js/zwave-js#5228] [Q&A] [answered]
Me and a colleague have set up an RPi3 with HA and Z-Wave JS. We are using an Aeotec Z-Stick Gen5+ as a gateway, in order to collect measurements from different sensors (a couple of Aeotec Multisensor 6 ones, an MCO Home A8-9 and a POPP Z-rain). The sensors have been added as devices through the Z-Wave JS absolutely fine (or at least it seems so). I’m able to receive data without a problem.

Now the strange part. Although I get measured values from the MCO Home (and I’m able to cross-check them, since it has an LCD screen), I just can’t get the CO2 level to be transferred correctly. It is always stuck to zero. Data are received per the required interval without a problem. I can see the Z-Wave JS logs that the value read is indeed zero. BUT, the sensor itself displays non-zero values (in fact, around 1500 ppm, as we are in an office with a lot of people). The strange thing is that the rest of the values are read and updated correctly. What I’ve tried so far:

-Adjust the settings in various ways (timers, delta thresholds etc.)
-Restart the sensor, the RPi, plug the Z-stick to another port

The only option that I have left is to delete the sensor from the network and re-configure it…but I can’t think how it will make a difference. Could be the sensor itself that has an issue? Has anyone else had a similar experience/worked with sensors from MCO Home in the past?

I attach the figures showing the problem and some of the Z-Wave JS logs.

![MCO Home A8-9 - Home assistant](https://user-images.githubusercontent.com/66252394/204277719-cb569ed7-96c2-44af-9f5a-74ef0f4906db.png)

![MCO Home A8-9 LCD1](https://user-images.githubusercontent.com/66252394/204277734-a444f918-ebfb-4a1a-a4cf-146e5ffb302c.jpg)

[Z-wave logs.txt](https://github.com/zwave-js/node-zwave-js/files/10103733/Z-wave.logs.txt)

**Comment by AlCalzone** [comment]
@robertsLando Don't you have the A8-9?

**Reply by robertsLando** [reply]
@AlCalzone nope mine it's MH5-2A

**Comment by nkatsaros443** [comment, marked as answer]
Well, problem solved: it is a hardware issue. MCOHome support sent us an updated STM firmware that needs to be added to the sensor through an in-circuit programmer. Now everything works fine.
